TransformationPlayground/playground_frontend/src/app/page.tsx

```diff
@@ -0,0 +1,206 @@
+"use client";
+
+import React, { useState } from "react";
+import "@cloudscape-design/global-styles/index.css";
+import { Grid, Container, FormField, Textarea, Button, Input, Select, SelectProps, Spinner } from "@cloudscape-design/components";
+
+const TransformationPage = () => {
+  // States for user inputs
+  const [inputShape, setInputShape] = useState("");
+  const [transformLogic, setTransformLogic] = useState("");
+  const [outputShape, setOutputShape] = useState("");
+  const [validationReport, setValidationReport] = useState("");
+  const [testTargetUrl, setTestTargetUrl] = useState("");
+  const [isLoading, setIsLoading] = useState(false);
+
+  // Select options for dropdowns
+  const sourceVersionOptions: SelectProps.Options = [{ label: "Elasticsearch 6.8", value: "Elasticsearch 6.8" }];
+  const targetVersionOptions: SelectProps.Options = [{ label: "OpenSearch 2.17", value: "OpenSearch 2.17" }];
+  const transformTypeOptions: SelectProps.Options = [{ label: "Index", value: "Index" }];
+  const transformLanguageOptions: SelectProps.Options = [{ label: "Python", value: "Python" }];
```

Comment:
Seems like this data should come from the API, or there should be a mapping of API data against the strings for visualization

Reply:
+1.  I think this comes down to a shared spec between the frontend and backend; this also applies to the shape of the requests/responses that are going over the wire.  I know we've briefly discussed OpenAPI or something as a model format.